src/hooks/usePermission.js

```diff
@@ -0,0 +1,24 @@
+import { getUser } from '../contexts'
+import { PERMISSION } from '../constants'
+
+const usePermission = () => {
+  const user = getUser()
+  const getPermission = (permission) => user?.permission?.includes(permission)
+
+  return {
+    canRead() {
+      return getPermission(PERMISSION.READ)
+    },
+    canCreate() {
+      return getPermission(PERMISSION.CREATE)
+    },
+    canUpdate() {
+      return getPermission(PERMISSION.UPDATE)
+    },
+    canDelete() {
+      return getPermission(PERMISSION.DELETE)
+    },
+  }
+}
```

Comment:
Look this function is not a hook, because it doesn't join the lifecycle of a component since not any react hooks are used here. I think we can name it `getPermission` and the `getPermission` function inside can rename to `checkPermission` because it returns a boolean. The new`getPermission` can move to the util folder.

Reply:
updated